activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeBPMNTimerIT.java

```diff
@@ -0,0 +1,292 @@
+        assertThat(eventsFired.size()).isEqualTo(0);
+        
+        
+        long waitTime = 50 * 60 * 1000;
+        Date dueDate = new Date(startTime.getTime() + waitTime);
+        
+        // After setting the clock to time '50minutes and 5 seconds', the second timer should fire
+        processEngineConfiguration.getClock().setCurrentTime(new Date(dueDate.getTime() + 5000));
+        waitForJobExecutorToProcessAllJobs(5000L, 25L);
+        
+        
+        //then
+        eventsFired = listenerFired.getEvents();
+         
+        assertThat(eventsFired)
+                .extracting(BPMNTimerEvent::getEventType,
+                            BPMNTimerEvent::getProcessDefinitionId,
+                            event -> event.getEntity().getProcessDefinitionId(),
+                            event -> event.getEntity().getTimerPayload().getJobType(),
+                            event -> event.getEntity().getTimerPayload().getJobHandlerType()         
+                )
+                .contains(Tuple.tuple(BPMNTimerEvent.TimerEvents.TIMER_FIRED,
+                                      process.getProcessDefinitionId(),
+                                      process.getProcessDefinitionId(),
+                                      "timer",
+                                      "trigger-timer"
+                                      )
+                     
+                );
+        
+        assertThat(listenerExecutionSuccess.getEvents())
+                .extracting(BPMNTimerEvent::getEventType,
+                            BPMNTimerEvent::getProcessDefinitionId,
+                            event -> event.getEntity().getProcessDefinitionId(),
+                            event -> event.getEntity().getTimerPayload().getJobType(),
+                            event -> event.getEntity().getTimerPayload().getJobHandlerType()         
+                )
+                .contains(Tuple.tuple(BPMNTimerEvent.TimerEvents.JOB_EXECUTION_SUCCESS,
+                                      process.getProcessDefinitionId(),
+                                      process.getProcessDefinitionId(),
+                                      "timer",
+                                      "trigger-timer"
+                                      )
+             
+        );
+ 
+        
+    }
+    
+    @Test
+    public void shouldGetTimerCanceledEventByManagementService() throws Exception {
+        // GIVEN
+        securityUtil.logInAs("salaboy");
+
+        ProcessInstance process = processRuntime.start(ProcessPayloadBuilder.start()
+                                                       .withProcessDefinitionKey(PROCESS_TIMER_CANCELED_EVENT)
+                                                       .build());
+        
+        String jobId = managementService.createTimerJobQuery().singleResult().getId();
+
+        assertThat(jobId).isNotNull();
+        
+        // WHEN
+        managementService.deleteTimerJob(jobId);
```

Comment:
Instead of deleting explicitly the job, couldn't we achieve the same result by deleting the process instance?

Reply:
The idea of the test was really to delete timerJob and to check events. But I may do another test and delete a process to check events...

Reply:
I don't think we need to cover this case directly. Covering process deleted is enough.

Reply:
Not needed test deleted: https://github.com/Activiti/Activiti/pull/2814/commits/d627bd328bdfddb7e96502e37e25e54d16b43766